about/process/branch-naming.rst

```diff
@@ -0,0 +1,54 @@
+Branch-naming convention
+========================
+
+Every repo's readme should state which branch-naming-convention is used and possible deviations from the norm.
+
+Click-Packages
+--------------
+
+Software that is exclusively distributed as a click-package (and not also as a deb) only uses one ``master`` branch that is protected. Separate temporary development branches with arbitrary descriptive names can be created and merged into master when the time comes. For marking and archiving milestones in development history, ideally Git tags or GitHub releases should be used.
+
+Deb-Packages
+------------
+
+To make most efficient use of our CI system, a special naming convention for git-branches is used.
+
+For pre-installed Ubuntu Touch components, deb-packages are used whereever possible. This includes coreapps, since they can still be independently updated using click-package downloads from the openstore. This policy allows us to make use of the powerful powerful debian build system to resolve dependencies.
+
+Every repository that uses this convention will have branches for the actively supported Ubuntu releases referenced by their codenames (``bionic``, ``xenial``, ``vivid``, etc.). These are the brances that are built directly into the corresponding images and published on :doc:`repo.ubports.com <ppa>`. If no separate versions for the different Ubuntu bases are needed, the repository will just have one ```master`` branch and the CI system will still still build versions for all actively supported releases and resolve dependencies accordingly.
+
+Branch-extensions
+^^^^^^^^^^^^^^^^^
+
+To build and publish packages based on a release, an extension in the form of  ``xenial_-_some-descriptive_extension`` can be used. The CI system will then resolve all dependencies using the ``xenial_-_some-descriptive_extension`` branch of other repositories or fall back on the normal ``xenial`` dependencies, if that doesn't exist. These special dependencies are not built into the image but still pushed to  on :doc:`repo.ubports.com <ppa>`.
```

Comment:
All dashes need to be surrounded by underscores, right? If so, you're missing that in a lot of places.

Reply:
No, only if you want to separate two stages of dependencies. Do you have an idea how i could make that clearer?

Reply:
I remember a document at some point that explained the creation of Jenkinsfiles and branch naming... It appears to be long gone. Maybe just a section explaining the reason for `_-_`?

Reply:
That would be either https://github.com/ubports/build-tools or the protocol of the meeting where we agreed on this structure. I don't think the reason why we made this decision is relevant or interesting to anyone, so i did not mention it.

The reason we use `_-_` is that we need a separator that is not usually used in a name of a stage, and other special characters are not an option. Thanks to `_-_`, you can still use `branch_names_extensions` `like-this` `or_even-this`. I thought that was implied, but if you think i should mention that in the document, i will add it.